airframe-sql/src/main/scala/wvlet/airframe/sql/analyzer/TypeResolver.scala

```diff
@@ -86,9 +86,12 @@ object TypeResolver extends LogSupport {
           case k @ GroupingKey(LongLiteral(i, _), _) if i <= selectItems.length =>
             // Use a simpler form of attributes
             val keyItem = selectItems(i.toInt - 1) match {
-              case a: Attribute =>
-                toResolvedAttribute(a.name, a)
-              case other => other
+              case SingleColumn(expr, _, _) =>
+                expr
+              case Alias(_, _, expr, _) =>
+                expr
+              case other =>
+                other
```

Comment:
Reverted but added unwrapping `Alias` to avoid wrong SQL generation in GROUP BY clause.